File: examples/experimental/scala-cleanup-app/src/main/scala/Serving.scala
package org.apache.predictionio.examples.experimental.cleanupapp

import org.apache.predictionio.controller.LServing

class Serving
  extends LServing[Query, PredictedResult] {

  override
  def serve(query: Query,
    predictedResults: Seq[PredictedResult]): PredictedResult = {
    predictedResults.head
  }
}

File: examples/experimental/java-local-regression/src/main/java/Serving.java
package org.apache.predictionio.examples.java.regression;

import org.apache.predictionio.controller.java.LJavaServing;
import org.apache.predictionio.controller.java.EmptyParams;

import java.lang.Iterable;

public class Serving extends LJavaServing<EmptyParams, Double[], Double> {
  public Double serve(Double[] query, Iterable<Double> predictions) {
    int n = 0;
    double s = 0.0;
    for (Double d: predictions) {
      n += 1;
      s += d;
    }
    return s / n;
  }
}
